docs/design/interface/pseudocode/check_data.py

```diff
@@ -0,0 +1,64 @@
+# ruff: noqa
+def check_data(
+    data: DataFrame, resource_properties: ResourceProperties
+) -> DataFrame | Error:
+    """Checks that the DataFrame matches the requirements in the resource properties.
+
+    Runs a few checks to compare between the data and the properties on the items:
+
+    | Data | Properties |
+    |:------|:------------|
+    | Column names | `field.names` |
```

Comment:
```suggestion
    | Column names | `field.name` |
```
I understand this means the names of all fields, but there is no `names` attribute. Same for type.

Reply:
Thanks for the catch! :grin: